src/interfaces/widgets/table.py

```diff
@@ -0,0 +1,139 @@
+"""
+
+    This module generate a table widget.
+
+"""
+
+from collections import OrderedDict
+
+from PyQt5.QtCore import Qt
+from PyQt5.QtWidgets import QTableWidget, QAbstractItemView, QTableWidgetItem
+
+
+
+class Table(QTableWidget):
+
+    current_row     : int = 0
+    count_row       : int = 0
+    count_column    : int = 0
+
+    """
+        :param widget: parent widget
+        :param count_row: max row in table
+        :param header: dictionary, key is the column label, value is the column width
+    """
+    def __init__(self, widget, count_row:int, header:OrderedDict, **kwargs):
+        super(Table, self).__init__(widget)
+
+        self.count_row = count_row
+
+        count_column = 0
+        header_label = []
+        header_width = []
+        for label, width in header.items():
```

Comment:
This can be shortened to 
```python
self.column_count = len(header)
header_label = [*header.keys()]
header_width = [*header.values()]
```

Or rather, you can just pass these values directly to `setHorizontalHeaderLabels` and `enumerate` Like so:
```python
self.setHorizontalHeaderLabels(header.keys())

for i, width in enumerate(header.values()):
    ...
```

Reply:
fixed.